chain/src/state_key.rs

```diff
@@ -27,3 +27,11 @@ pub fn halt_now() -> &'static str {
 pub fn chain_params_changed() -> &'static str {
     "chain_params_changed"
 }
+
+pub fn epoch_by_height(height: u64) -> String {
+    format!("chain/epoch_by_height/{}", height)
+}
```

Comment:
I started wondering about how we could avoid storing a height => epoch lookup for every height, but I think that the current storage APIs are a bit limiting, in that they only give us prefix queries rather than real range operations (even though those operations could be implemented). So this seems reasonable to do in the meantime.